rfd/00TBD-webui-paginate-filter.md

```diff
@@ -0,0 +1,335 @@
+| `limit`            | Maximum number of rows to return on each fetch.                                               |
+| `startKey`         | Resume row search from the last row received. Empty string means start search from beginning. |
+| `q`                | Start of filter query.                                                                        |
+
+<br/>
+
+| Filter Identifier | Description                                                       |
+|-------------------|-------------------------------------------------------------------|
+| `l=<key>:<value>` | Identifies `<key>:<value>` as a label filter.                     |
+| `s=<col>:<dir>`   | Identifies `<col>:<dir>` as a sort filter.                        |
+| `<value>`         | Values with no identifiers will be interpreted as a search value. |
+
+<br/>
+
+| Delimiters  | Description                                                                                 |
+|-------------|---------------------------------------------------------------------------------------------|
+| `:` (colon) | Delimiter in a filter identifier to mean separate values.                                   |
+| `+` (plus)  | A filter delimiter `AND` to mean inclusion of filter values. Only used in URL.              |
+| ` ` (space) | A filter delimiter `AND` to mean inclusion of filter values. Only used in CLI or search bar |
+| `-` (minus) | A filter delimiter `NOT` to mean exclusion of filter value.                                 |
+| `,` (comma) | A filter delimiter `OR` to mean match any filter values.                                    |
+
+<br/>
+
+**Examples**
+
+| URL Query Examples                  | Description                                                                    |
+|-------------------------------------|--------------------------------------------------------------------------------|
+| `?q=l=env:prod+l=country:US`        | rows must contain labels `env:prod` and `country:US`                           |
+| `?q=l=os:mac,os:linux+-l=env:prod`  | rows can contain labels `os:mac` or `os:linux` but not `env:prod`              |
+| `?q=-l=os:mac+banana`               | rows cannot contain label `os:mac` but contain search value `banana`           |
+| `?q=l=os:mac+s=colName:desc+banana` | rows must contain `os:mac` and `banana` and sorted by column `name` descending |
+
+<br/>
+
+| User Query Equivalent                    | Description                                                                    |
+|------------------------------------------|--------------------------------------------------------------------------------|
+| `tsh ls l=env:prod l=country:US`         | rows must contain labels `env:prod` and `country:US`                           |
+| `tsh ls l=os:mac,l=os:linux -l=env:prod` | rows can contain labels `os:mac` or `os:linux` but not `env:prod`              |
```

Comment:
This would silently break backwards-compatibility, since `tsh ls` already interprets `<key>=<val>` as a label.  Meaning that existing versions would interpret this is `"l"="env:prod"` and behave strangely.